packages/flutter/lib/src/material/focus_highlight.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// A widget that shows a highlight decoration around its child when a
+/// [Focus] above it in the widget hierarchy is focused.
+///
+/// It allows specification of a [focusedDecoration] to be shown when it
+/// has focus, and [unfocusedDecoration] when it is not.
+///
+/// By default, it uses the values derived from [Theme.focusHighlightTheme].
+class FocusHighlight extends StatefulWidget {
+  /// Creates a widget that manages a [FocusNode]
+  ///
+  /// The [child] argument is required and must not be null.
+  ///
+  /// The [autofocus], and [showDecorations] arguments must not be null.
+  const FocusHighlight({
+    Key key,
+    @required this.child,
+    this.focusedDecoration,
+    this.unfocusedDecoration,
+    this.duration,
+    this.curve,
+  })  : assert(child != null),
+        super(key: key);
+
+  /// The child widget of this [FocusHighlight].
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// Defines the decoration to be drawn around the [child] when this widget
+  /// receives the focus.
+  ///
+  /// This defaults to the [FocusHighlightThemeData.focusedDecoration] provided by
+  /// the ambient [Theme] in the context.
+  final Decoration focusedDecoration;
+
+  /// Defines the decoration to be drawn around the [child] when this widget
+  /// does not have the focus.
+  ///
+  /// This defaults to the [FocusHighlightThemeData.unfocusedDecoration] provided by
+  /// the ambient [Theme] in the context.
+  final Decoration unfocusedDecoration;
+
+  /// The duration over which to animate the parameters of the focus highlight
+  /// [decoration].
+  final Duration duration;
+
+  /// The curve to apply when animating the parameters of the focus highlight
+  /// [decoration].
+  final Curve curve;
+
+  @override
+  _FocusHighlightState createState() => _FocusHighlightState();
+}
+
+class _FocusHighlightState extends State<FocusHighlight> {
+  bool _hasFocus;
+  FocusNode _node;
+
+  @override
+  void didUpdateWidget(FocusHighlight oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    final FocusNode newNode = Focus.of(context);
+    if (newNode != _node) {
+      _node.removeListener(_handleFocusChanged);
+      _node = newNode;
+      _node.addListener(_handleFocusChanged);
+    }
```

Comment:
this looks like what i'd expect to see in didChangeDependencies rather than didUpdateWidget